rust/agama-software/src/model.rs

```diff
@@ -143,11 +127,23 @@ impl ModelAdapter for Model {
 
     /// Returns the software system information.
     async fn system_info(&self) -> Result<SystemInfo, service::Error> {
-        Ok(SystemInfo {
-            patterns: self.patterns().await?,
-            repositories: vec![],
-            addons: vec![],
-        })
+        let Some(product_spec) = self.selected_product.clone() else {
+            return Err(service::Error::MissingProduct);
+        };
+
+        let (tx, rx) = oneshot::channel();
+        self.zypp_sender
+            .send(SoftwareAction::GetSystemInfo(product_spec, tx))?;
+        let mut system_info = rx.await??;
+
+        // set "mandatory" field as the info to decide whether the repository is mandatory or not
+        // lives in this struct.
```

Comment:
I am not sure if this comment is not more confusing. I would write here simply that "set as mandatory local repositories".

Reply:
yeap, better name will help for sure...maybe "predefined"? Like that it is not the ones coming from user?

Reply:
I like predefined. I will adapt the code.